conbench/app/_plots.py

```diff
@@ -237,6 +238,28 @@ def time_series_plot(history, benchmark, run, height=250, width=1000):
         legend_label="Benchmark",
         name="benchmark",
     )
+
+    step_changes = [
+        Span(
+            location=dateutil.parser.isoparse(result["timestamp"]),
+            dimension="height",
+            line_color="black",
+        )
+        for result in history
+        if result["is_step_change"]
+    ]
+    for step_change in step_changes:
+        p.add_layout(step_change)
+    if step_changes:
+        # hack: add a dummy line so it appears on the legend
+        one_result = [result for result in history if result["is_step_change"]][0]
+        p.line(
+            [dateutil.parser.isoparse(one_result["timestamp"])] * 2,
+            [one_result["mean"]] * 2,
+            legend_label="Step change",
```

Comment:
This really deserves a fuller explanation somewhere explaining the definition here in terms of z-scores and what the threshold is, etc. Maybe in a hover if we can?

Reply:
I'm sorta leaving this vague on purpose for now -- I think we need to collect more data before we can make a solid recommendation on how to use the `step_change` field. We can definitely come back and add a better definition once we do, e.g., #476.

Reply:
While I agree that gathering data is really great, we still will need to nudge folks in the right direction in order for them to use it at all (such that we would have any substantive points labelled as step-changes).

We can be vague about the details if we think that's helpful but something along the lines of "Mark this as a large shift in the data, which should be treated separately from the data before it for the purposes of variability measurement"

Reply:
That makes sense to me! I'll update the description by the checkbox

Reply:
What do you think of this as hover-text on the button?
<img width="449" alt="Screenshot 2023-01-06 at 14 37 49" src="https://user-images.githubusercontent.com/16600275/211096312-ec08c2a5-6e6a-4434-8977-1ecd56912c52.png">

Reply:
Update: 
<img width="626" alt="Screenshot 2023-01-10 at 16 08 12" src="https://user-images.githubusercontent.com/16600275/211673367-ba657ea0-c294-414b-a8e4-cebcbcf9e6b6.png">

Reply:
Maybe pedantic, sorry, but I don't love "permanent" as a descriptor because inevitably there will be more changes in the future. "Durable" maybe?

Regardless, do we want users to not click this button for spikes, or quick deviations for a few commits that revert? If they're because of issues that resolve, clicking seems good in terms of getting them healthy distributions (and therefore future alerting). If we want to keep outliers separate that's ok, but I'm not convinced we need to here—though this is definitely the sort of thing it'd be useful to collect free text about as we figure out how to evolve our handling here.

Reply:
> If we want to keep outliers separate that's ok, but I'm not convinced we need to here—though this is definitely the sort of thing it'd be useful to collect free text about as we figure out how to evolve our handling here.

Yeah agreed, I would also strongly advocate for bundling "purposeful change that does indeed persist" with "this outlier is a blip that I can't explain which goes down immediately after" together _for now_. We'll follow up with making that distinction later when we have more real info if any of this is helpful + we have real data we can work with.

Maybe we could call this just "a change in distribution" instead of qualifying it. It's a bit ambiguous, but we might want to be purposefully so for now (and we have the hover over where we can explain that in a bit more detail.

Reply:
Yep, this makes sense, and ties into the fact that we can use the new `change_annotations` for anything. Let's try it.

Reply:
Okay, changed everything to "distribution change."